File: src/co/soundwave/repository/album/AlbumDAO.kt
package co.soundwave.repository.album

import co.soundwave.models.Album
import co.soundwave.repository.BaseDAO

class AlbumDAO : BaseDAO<AlbumSQL, Album>() {
    override val sql: AlbumSQL by lazy { AlbumSQL() }
    fun getInsertQuery(obj: Album, artistId: Int): String =
        sql.insertAlbum(obj.name, obj.imgPath, obj.releaseYear, artistId)
}

File: repository/album/AlbumDAO.py
from models import Album as alb
from repository import BaseDAO as bdao
from repository.album import AlbumSQL as alsql


class AlbumDAO(bdao.BaseDAO):
    def __init__(self):
        super().__init__()
        self._sql = alsql.AlbumSQL()

    def get_insert_query(self, album: alb.Album):
        return alsql.AlbumSQL.insert_album(album.name, album.img_path,
                                           album.release_year, album.artist_id)
